server/src/api/modules/players/services/UpdatePlayerService.ts

```diff
@@ -26,10 +26,18 @@ async function updatePlayer(username: string, skipFlagChecks = false): Promise<U
   // Find a player with the given username or create a new one if needed
   const [player, isNew] = await findOrCreate(username);
 
+  if (player.annotations?.some(a => a.type === PlayerAnnotationType.OPT_OUT)) {
+    throw new ForbiddenError('Failed to update: Player has opted out.');
+  }
+
+  if (player.annotations?.some(a => a.type === PlayerAnnotationType.BLOCKED)) {
+    //contact disc
```

Comment:
`throw new ForbiddenError("This player has been blocked, please contact us on Discord for more information.")`